### PR TITLE
run_node doc improvements

### LINKED ISSUE #51: Question: invariance to evaluation interval
Consider the following example:
```julia
using TimeDag
node = TimeDag.block_node(Block(DateTime(2020, 01, 01):Day(1):DateTime(2020, 01, 05), ones(5)))
v = sum(node)
```
and evaluation over two different time windows:
```julia
julia> evaluate(v, DateTime(2020, 01, 01), DateTime(2020, 01, 06))
Block{Float64}(5 knots)
|                time |   value |
|            DateTime | Float64 |
|---------------------|---------|
| 2020-01-01T00:00:00 |     1.0 |
| 2020-01-02T00:00:00 |     2.0 |
| 2020-01-03T00:00:00 |     3.0 |
| 2020-01-04T00:00:00 |     4.0 |
| 2020-01-05T00:00:00 |     5.0 |


julia> evaluate(v, DateTime(2020, 01, 02), DateTime(2020, 01, 06))
Block{Float64}(4 knots)
|                time |   value |
|            DateTime | Float64 |
|---------------------|---------|
| 2020-01-02T00:00:00 |     1.0 |
| 2020-01-03T00:00:00 |     2.0 |
| 2020-01-04T00:00:00 |     3.0 |
| 2020-01-05T00:00:00 |     4.0 |
```

The docstring for `TimeDag.run_node!` contains the following remark:
```
Consistency: Calling run_node! over a single interval should give the same result as calling it multiple times over a decomposition of that same interval.
```
@tpgillam Could you elaborate on what is meant here? To me, the above looks like an example of this being violated, in the sense that e.g. the value associated with `2020-01-02T00:00:00` is different depending upon which time range we evaluate (which feels like an unavoidable property of cumulative sumation)

### PR DESCRIPTION
**Pull request by willtebbutt**
Closes #51

### CODE REVIEW

File: docs/src/reference/creating_ops.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ In order to create a source — i.e. an op with zero inputs — one should use t
 
 ## Low-level API
 
-The most general way to create an op is to create a structure that inherits from [`TimeDag.NodeOp`](@ref), and implement [`TimeDag.run_node!`](@ref).
+The most general way to create an op is to create a structure that inherits from [`TimeDag.NodeOp`](@ref), and implements [`TimeDag.run_node!`](@ref) and [`TimeDag.create_evaluation_state`](@ref).
```
Comment (tpgillam) [inline docs/src/reference/creating_ops.md:22]:
```suggestion
The most general way to create an op is to create a structure that inherits from [`TimeDag.NodeOp`](@ref), and implements both of:
* [`TimeDag.run_node!`](@ref) 
* [`TimeDag.create_evaluation_state`](@ref), OR defines [`TimeDag.stateless`](@ref) to be true.
```

Comment (tpgillam) [reply docs/src/reference/creating_ops.md:22]: And actually I think `TimeDag.stateless` needs to be added to the list of functions for which docs should be generated. Probably immediately after `TimeDag.create_evaluation_state` in `internals.md`

**Review by tpgillam — APPROVED** [review]
LGTM subject to minor alterations!

**Comment by willtebbutt** [comment]
Excellent. Have incorporated suggestions. Will merge when CI passes

**Pull request merged**